pop-api/examples/fungibles/tests.rs

```diff
@@ -0,0 +1,884 @@
+use drink::{
+	assert_err, assert_ok, call,
+	devnet::{
+		account_id_from_slice,
+		error::{
+			v0::{ApiError::*, ArithmeticError::*, Error},
+			Assets,
+			AssetsError::*,
+		},
+		AccountId, Balance, Runtime,
+	},
+	last_contract_event,
+	session::Session,
+	AssetsAPI, TestExternalities, NO_SALT,
+};
+use ink::scale::Encode;
+use pop_api::{
+	primitives::TokenId,
+	v0::fungibles::events::{Approval, Created, Transfer},
+};
+
+use super::*;
+
+const UNIT: Balance = 10_000_000_000;
+const INIT_AMOUNT: Balance = 100_000_000 * UNIT;
+const INIT_VALUE: Balance = 100 * UNIT;
+const ALICE: AccountId = AccountId::new([1u8; 32]);
+const BOB: AccountId = AccountId::new([2_u8; 32]);
+const CHARLIE: AccountId = AccountId::new([3_u8; 32]);
+const AMOUNT: Balance = MIN_BALANCE * 4;
+const MIN_BALANCE: Balance = 10_000;
+const TOKEN: TokenId = 1;
+
+// The contract bundle provider.
+//
+// See https://github.com/r0gue-io/pop-drink/blob/main/drink/test-macro/src/lib.rs for more information.
+#[drink::contract_bundle_provider]
+enum BundleProvider {}
+
+/// Sandbox environment for Pop Devnet Runtime.
+pub struct Pop {
+	ext: TestExternalities,
+}
+
+impl Default for Pop {
+	fn default() -> Self {
+		// Initialising genesis state, providing accounts with an initial balance.
+		let balances: Vec<(AccountId, u128)> =
+			vec![(ALICE, INIT_AMOUNT), (BOB, INIT_AMOUNT), (CHARLIE, INIT_AMOUNT)];
+		let ext = BlockBuilder::<Runtime>::new_ext(balances);
+		Self { ext }
+	}
+}
+
+// Implement core functionalities for the `Pop` sandbox.
+drink::impl_sandbox!(Pop, Runtime, ALICE);
+
+// Deployment and constructor method tests.
+
+#[drink::test(sandbox = Pop)]
+fn new_constructor_works(mut session: Session) {
+	let _ = env_logger::try_init();
+	// Deploy a new contract.
+	let contract =
+		deploy(&mut session, "new", vec![TOKEN.to_string(), MIN_BALANCE.to_string()]).unwrap();
+	// Token exists after the deployment.
+	assert!(session.sandbox().asset_exists(&TOKEN));
+	// Successfully emit event.
+	assert_eq!(
+		last_contract_event(&session).unwrap(),
+		Created {
+			id: TOKEN,
+			creator: account_id_from_slice(&contract),
+			admin: account_id_from_slice(&contract),
+		}
+		.encode()
+		.as_slice()
+	);
```

Comment:
This can be improved but not now. Opened an issue: https://github.com/r0gue-io/pop-drink/issues/24

Reply:
Wouldn't this improve these assertions a lot:
```
#[ink::event]
#[cfg_attr(feature = "std", derive(Debug, PartialEq))]
pub struct Transfer {
	/// The source of the transfer. `None` when minting.
	#[ink(topic)]
	pub from: Option<AccountId>,
	/// The recipient of the transfer. `None` when burning.
	#[ink(topic)]
	pub to: Option<AccountId>,
	/// The amount transferred (or minted/burned).
	pub value: u128,
}
```

Is working in the integration tests for me. This gives much clearer errors.

Reply:
Resolved in https://github.com/r0gue-io/pop-node/pull/370